pkg/imageregistry/imageregistry.go

```diff
@@ -134,8 +134,23 @@ func (builder *Builder) GetManagementState() (v1.ManagementState, error) {
 	return builder.Object.Spec.ManagementState, nil
 }
 
+// GetStorageConfig fetches imageRegistry Storage configuration.
+func (builder *Builder) GetStorageConfig() (imageregistryv1.ImageRegistryConfigStorage, error) {
+	if valid, err := builder.validate(); !valid {
+		return imageregistryv1.ImageRegistryConfigStorage{}, err
```

Comment:
return nil, err

Reply:
I prefer to leave it like this

Reply:
nevermind, I'll use a pointer on comparing. I don't like personally how usage of this result is look like. But let's make some progress with this commit. I'm blocked with the other code I have